docs/generate_tables/xco-html.xqm

```diff
@@ -0,0 +1,1056 @@
+    let $ftree := fu:ftree($relPaths, $dirPath)
+    let $linkTree := hl:contabReportIndex_tocTree($ftree, $domainDict, $options)
+    let $title := 'Table of Contents'
+    let $subtitle :=        
+        let $reportLabel := 
+            if ($options?domainType eq 'xsd') then 'XSD reports' else 'reports'
+        return count($domains)||' '||$reportLabel
+    let $_DEBUG := dg:WRITE_FILE($ftree, 'FTREE.XML', 'ftree', 'yes', $options)
+    return
+    
+    <div class="toc" id="toc">
+        <div id="toctitle">
+            <h4>{$title}</h4>
+            <p class="tocsubtitle">{$subtitle}</p>
+        </div>
+        <ul class="sectlevel1">{
+          $linkTree
+        }</ul>
+    </div>
+};
+
+(:~
+ : Maps a folder tree representation of report files to
+ : a sequence of list item elements representing folders
+ : and files
+ :
+ : @param ftree folder tree 
+ : @param domainDict maps paths to domain information
+ : @param options options controlling the processing
+ :)
+declare function hl:contabReportIndex_tocTree(
+                              $ftree as element(fo),
+                              $domainDict as map(xs:string, item()*),
+                              $options as map(xs:string, item()*)?)
+        as element(li)* {
+    hl:contabReportIndex_tocTreeREC($ftree, $domainDict, $options)
+};
+
+declare function hl:contabReportIndex_tocTreeREC(
+                              $n as node(),
+                              $domainDict as map(xs:string, item()*),
+                              $options as map(xs:string, item()*)?)
+        as element()* {
+    typeswitch($n)
+    case element(fo) return
+        let $level := count($n/ancestor::fo)
+        let $prefix := (for $i in 1 to $level return '>> ') => string-join('')
+        return (
+            <li>{
+                <span class="monospace">{$prefix}</span>,
+                <span class="darkbrown">{$n/@name/string()}</span>
+            }</li>,
+            $n/* ! hl:contabReportIndex_tocTreeREC(., $domainDict, $options)
+        )
+    case element(fi) return
+        let $href := $n/ancestor-or-self::*/@name => string-join('/')    
+        let $isEnumDict := contains($href, 'enum-dict')
+        return if ($isEnumDict and not($options?withEnumDict)) then () else
+        
+        let $path := $n/ancestor-or-self::*/@name => string-join('/')
+        let $level := count($n/ancestor::fo)
+        let $prefix := (for $i in 1 to $level return '.  ') => string-join('')
+        let $dinfo := $domainDict($path)
+        let $href := $dinfo?relPath
+        let $title := $dinfo?fileName||
+                      ($dinfo?countEnumTypes[string()] ! (' ('||.||')'))
+        let $titleClass := attribute class {'enum-dict'} 
+                           [contains($href, 'enum-dict')]
+        return
+            <li>{
+                <span class="monospace">{$prefix}</span>,
+                <a shape="rect" href="{$href}">{$titleClass, $title}</a>
```

Comment:
Instead of a simple filename `ojp.html` the list should contain the title of the documents linking to that file.
https://vdvde.github.io/OJP/feature/documentation-tables-xcore/documentation-tables/index.html

After all this is the starting page for our documentation.